doc/development_methodology.rst

```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
 
 Release lifecycle
 -----------------
-A release is started by forking from the top of **devel**. The first commit added to the release branch bumps the release version. From that moment, all the bugfixes relevant to the current release must be merged only into the corresponding release branch. No new features could be added to it.
 
-When the release is ready, it is merged into **master** with the suitable release tag. Next it is merged into the current **devel** branch to integrate the bugfixes. If a bugfix from the current release branch is really, really necessary for the current **devel** branch, it could be cherry-picked onto **devel**. But be aware that each of these cherry-pickings eventually results in a merge conflict that should be resolved manually.
+We follows `SemVer <https://semver.org/>`_ notation for release naming.
 
-The release branch lives in the Tribler repository for as long as we support the corresponding Tribler version. Eventually, the branch is removed.
+* Stable release name: <MAJOR>.<MINOR>.<PATCH>
+* Release candidate name: <MAJOR>.<MINOR>.<PATCH>-RC<NUMBER>
+
+The release lifecycle:
+
+1. A release branch forks from `devel`
+2. The release branch is checked by running the application-tester
+3. In case no errors, a release candidate (RC) is published as pre-release in Github and published to the forum as release post.
+4. After a while (1-2 weeks) release candidate become a stable
+5. The stable release is delivered to users
+6. The release branch is merged to `devel`
+
+
+In case of an error than needs to be fixed asap in the published stable release:
+
+1. A fix is committed to the corresponding release branch
+2. The release branch is checked by running the application-tester
+3. The fixed release is delivered to users
+4. The release branch is merged to `devel`
+
+
+A release is started by forking from the top of **devel**.
+From that moment, all the bugfixes relevant to the current release must be merged into the corresponding release branch.
+No new features could be added to it.
+
+When the release is ready to publishing, it is merged back to **devel** to integrate the bugfixes.
```

Comment:
Suggestion:
to publishing -> for publishing

Reply:
Fixed